src/main/java/com/commercetools/sync/products/helpers/ProductReferenceResolver.java

```
@@ -147,10 +147,10 @@ public ProductReferenceResolver(@Nonnull final ProductSyncOptions productSyncOpt
             return resolveResourceIdentifier(draftBuilder, draftBuilder.getProductType(),
                 productTypeService::fetchCachedProductTypeId, ResourceIdentifier::ofId,
                 ProductDraftBuilder::productType);
-        } catch (ReferenceResolutionException referenceResolutionException) {
+        } catch (Exception exception) {
```

That does not seem right to me. The catch block is only handling the reference resolution exception, maybe we would need another handler for generic Exception, instead of swallowing it.

---

This is the purpose. ReferenceResolutionException is considered as exception wrapper to indicate any exception thrown during resolveReference(), thus any unchecked exception occurs in runtime should be included.

---

if it's not ReferenceResolutionException but let's say NPE (null pointer exception)? IMO, you added that part, but handling the exception is not changed, thus changing only catch block is not enough.

---

This change has been rolled back.